core/src/main/scala/org/http4s/blaze/pipeline/stages/OneMessageStage.scala

```diff
@@ -0,0 +1,27 @@
+package org.http4s.blaze.pipeline.stages
+
+import org.http4s.blaze.pipeline.MidStage
+
+import scala.concurrent.Future
+
+/** Holds a single element that when read, will eject itself
+  * from the pipeline.
+  * @note There is an intrinsic race condition between this stage
+  *       removing itself and write commands, so this stage should
+  *       only be used under half duplex conditions.
```

Comment:
I'm sorry, I don't understand. Enforce the half duplex requirement? Maybe I used the wrong terminology, but I meant that they shouldn't be used concurrently. Maybe I should rephrase it.